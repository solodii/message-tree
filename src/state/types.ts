export interface Message {
  id: string;
  subjectId: string;
  date: Date;
  text: string;
}

export interface Draft {
  subjectId: string;
  text: string;
}

export interface MessagesById {
  [id: string]: Message;
}

export interface MessagesBySubject {
  [subjectId: string]: Message[];
}

export interface DraftsBySubject {
  [subjectId: string]: Draft;
}

export interface State {
  draftsBySubject: DraftsBySubject;
  messagesById: MessagesById;
  messagesBySubject: MessagesBySubject;
}

export interface MessageTree {
  message: Message;
  children: MessageTree[];
  hasDraft: boolean;
}
