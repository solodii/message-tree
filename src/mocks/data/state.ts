import { State } from '../../state/types';
import { draftsBySubject } from './drafts';
import {
  messagesById,
  messagesBySubject
} from './messages';

export const state: State = {
  draftsBySubject,
  messagesById,
  messagesBySubject
};
