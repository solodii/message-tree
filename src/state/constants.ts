export const ROOT_SUBJECT = '';
