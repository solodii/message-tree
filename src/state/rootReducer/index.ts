import { combineReducers } from 'redux';
import { State } from '../types';
import draftsBySubject from './draftsBySubject';
import messagesById from './messagesById';
import messagesBySubject from './messagesBySubject';

export default combineReducers<State>({
  draftsBySubject,
  messagesById,
  messagesBySubject
});
