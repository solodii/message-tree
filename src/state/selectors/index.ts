export * from './drafts';
export * from './messages';
