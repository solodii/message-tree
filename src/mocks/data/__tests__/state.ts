import { state } from '../state';

test('state', () => {
  expect(state).toMatchSnapshot();
});
