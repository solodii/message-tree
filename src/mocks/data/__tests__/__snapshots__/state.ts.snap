// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state 1`] = `
Object {
  "draftsBySubject": Object {
    "": Object {
      "subjectId": "",
      "text": "",
    },
    "0a6a7bfd-a212-4ae1-8125-6c7486213a94": Object {
      "subjectId": "0a6a7bfd-a212-4ae1-8125-6c7486213a94",
      "text": "draft for 0a6a7bfd-a212-4ae1-8125-6c7486213a94",
    },
  },
  "messagesById": Object {
    "0a6a7bfd-a212-4ae1-8125-6c7486213a94": Object {
      "date": 2016-11-10T08:01:00.000Z,
      "id": "0a6a7bfd-a212-4ae1-8125-6c7486213a94",
      "subjectId": "",
      "text": "first",
    },
    "6c6f1976-ff5b-420b-bc9d-f7484f0ee53f": Object {
      "date": 2016-11-10T08:02:00.000Z,
      "id": "6c6f1976-ff5b-420b-bc9d-f7484f0ee53f",
      "subjectId": "",
      "text": "second",
    },
    "913d97b3-6e04-4c1e-a11c-222221abf00e": Object {
      "date": 2016-11-10T08:03:00.000Z,
      "id": "913d97b3-6e04-4c1e-a11c-222221abf00e",
      "subjectId": "0a6a7bfd-a212-4ae1-8125-6c7486213a94",
      "text": "third",
    },
  },
  "messagesBySubject": Object {
    "": Array [
      Object {
        "date": 2016-11-10T08:01:00.000Z,
        "id": "0a6a7bfd-a212-4ae1-8125-6c7486213a94",
        "subjectId": "",
        "text": "first",
      },
      Object {
        "date": 2016-11-10T08:02:00.000Z,
        "id": "6c6f1976-ff5b-420b-bc9d-f7484f0ee53f",
        "subjectId": "",
        "text": "second",
      },
    ],
    "0a6a7bfd-a212-4ae1-8125-6c7486213a94": Array [
      Object {
        "date": 2016-11-10T08:03:00.000Z,
        "id": "913d97b3-6e04-4c1e-a11c-222221abf00e",
        "subjectId": "0a6a7bfd-a212-4ae1-8125-6c7486213a94",
        "text": "third",
      },
    ],
  },
}
`;
